scapy/arch/windows/__init__.py

```diff
@@ -177,24 +177,40 @@ def _vbs_exec_code(code, split_tag="@"):
             yield l
     os.unlink(tmpfile.name)
 
-def _vbs_get_hardware_iface_guid(devid):
+def _get_hardware_iface_guid(devid):
+
+    devid = int(devid) + 1
+
+    hkey = winreg.HKEY_LOCAL_MACHINE
+    node = r"SOFTWARE\Microsoft\Windows NT\CurrentVersion\NetworkCards\{}".format(devid)
+    try:
+        key = winreg.OpenKey(hkey, node)
+        guid, _ = winreg.QueryValueEx(key, "ServiceName")
+        winreg.CloseKey(key)
+    except WindowsError:
+        return None
+    guid = guid[:-1] if guid.endswith("}\n") else guid
```

Comment:
Is'nt `guid.strip()` enough?

Reply:
Definitely yes, but I wanted to keep the previous logic strictly.
I will change it in next commit.